src/main/java/com/spotinst/sdkjava/model/ISpotAzureStatefulNodeRepo.java

```diff
@@ -0,0 +1,74 @@
+package com.spotinst.sdkjava.model;
+
+import com.spotinst.sdkjava.exception.SpotinstNotSupportedException;
+import com.spotinst.sdkjava.model.bl.azure.statefulNode.*;
+import com.spotinst.sdkjava.model.requests.azure.statefulNode.StatefulNodeAttachDataDiskRequest;
+import com.spotinst.sdkjava.model.requests.azure.statefulNode.StatefulNodeDetachDataDiskRequest;
+import com.spotinst.sdkjava.model.requests.azure.statefulNode.StatefulNodeGetLogsRequest;
+import com.spotinst.sdkjava.model.requests.azure.statefulNode.StatefulNodeStateChangeRequest;
+
+import java.util.List;
+
+public interface ISpotAzureStatefulNodeRepo extends IRepository<StatefulNode, Void, String>{
+
+    default RepoGenericResponse<StatefulNode> createNode(StatefulNode statefulNode, String authToken, String account) {
+        throw new SpotinstNotSupportedException();
+    }
+
+    default RepoGenericResponse<StatefulNode> getNode(String nodeId, String authToken, String account){
+        throw new SpotinstNotSupportedException();
+    }
+
+    default RepoGenericResponse<StatefulNode> updateNode(StatefulNode statefulNodeToUpdate, String nodeId, String authToken, String account){
+        throw new SpotinstNotSupportedException();
+    }
+
+    default RepoGenericResponse<Boolean> deleteNode(StatefulNodeDeallocationConfig statefulNodeDeallocationConfig, String nodeId, String authToken, String account){
+        throw new SpotinstNotSupportedException();
+    }
+
+    default RepoGenericResponse<List<StatefulNode>> getAllNodes(String authToken, String account){
+        throw new SpotinstNotSupportedException();
+    }
+
+    default RepoGenericResponse<Boolean> updateNodeState(StatefulNodeStateChangeRequest recycleStatefulNodeRequest, String nodeId, String authToken, String account){
```

Comment:
Change variable name "recycleStatefulNodeRequest" to something generic, not specific to recycle